### PR TITLE
Fixed a logic error in GetDuplicatePokemonToTransfer & reduced size of GetXpDiff

### PR DESCRIPTION
**Pull request by 5andr0**
With KeepPokemonsThatCanEvolve=true it also kept all pokemons that couldn't evolve. Added KeepPokemonsThatCanEvolve to Settings

### CODE REVIEW

File: PoGo.NecroBot.Logic/Inventory.cs [hunk 1]
```diff
@@ -83,13 +83,11 @@ public Inventory(Client client, LogicClient logicClient)
                 {
                     var settings = pokemonSettings.Single(x => x.PokemonId == pokemon.Key);
                     var familyCandy = pokemonFamilies.Single(x => settings.FamilyId == x.FamilyId);
-                    if (settings.CandyToEvolve == 0)
-                        continue;
+                    var amountToSkip = _client.Settings.KeepMinDuplicatePokemon;
+
+                    if (settings.CandyToEvolve > 0 && familyCandy.Candy / settings.CandyToEvolve > amountToSkip)
```
Comment (yamashi) [inline PoGo.NecroBot.Logic/Inventory.cs]: extract familyCandy.Candy / settings.CandyToEvolve out as you use it twice.

Comment (5andr0) [inline PoGo.NecroBot.Logic/Inventory.cs]: sry i thought i'd rather save a line..

File: PoGo.NecroBot.Logic/Inventory.cs [hunk 2]
```diff
@@ -83,13 +83,12 @@ public Inventory(Client client, LogicClient logicClient)
                 {
                     var settings = pokemonSettings.Single(x => x.PokemonId == pokemon.Key);
                     var familyCandy = pokemonFamilies.Single(x => settings.FamilyId == x.FamilyId);
-                    if (settings.CandyToEvolve == 0)
-                        continue;
+                    var amountToSkip = _logicClient.Settings.KeepMinDuplicatePokemon;
+                    var amountPossible = familyCandy.Candy / settings.CandyToEvolve;
```
Comment (mijanx) [inline PoGo.NecroBot.Logic/Inventory.cs]: Slight snag with a possible divide by zero here by moving it out like this.

Comment (yamashi) [inline PoGo.NecroBot.Logic/Inventory.cs]: Right the condition that was removed should not have been removed

Comment (5andr0) [inline PoGo.NecroBot.Logic/Inventory.cs]: Thanks @mijanx
Was a little hasty moving it out. Fixed here: [508](https://github.com/NecronomiconCoding/NecroBot/pull/508)

**Pull request merged**